src/main/io/osd.c

```diff
@@ -666,32 +669,33 @@ static bool osdDrawSingleElement(uint8_t item)
                 const armingDisableFlags_e flags = getArmingDisableFlags();
                 for (int i = 0; i < ARMING_DISABLE_FLAGS_COUNT; i++) {
                     if (flags & (1 << i)) {
-                        osdFormatMessage(buff, sizeof(buff), armingDisableFlagNames[i]);
+                        osdFormatMessage(buff, OSD_WARNINGS_MAX_SIZE + 1, armingDisableFlagNames[i]);
                         break;
                     }
                 }
                 break;
             }
 
             if (enabledWarnings & OSD_WARNING_BATTERY_WARNING && batteryState == BATTERY_WARNING) {
-                osdFormatMessage(buff, sizeof(buff), "LOW BATTERY");
+                osdFormatMessage(buff, OSD_WARNINGS_MAX_SIZE + 1, "LOW BATTERY");
                 break;
             }
 
             // Show warning if battery is not fresh
             if (enabledWarnings & OSD_WARNING_BATTERY_NOT_FULL && !ARMING_FLAG(WAS_EVER_ARMED) && (getBatteryState() == BATTERY_OK)
                   && getBatteryAverageCellVoltage() < batteryConfig()->vbatfullcellvoltage) {
-                osdFormatMessage(buff, sizeof(buff), "BATT NOT FULL");
+                osdFormatMessage(buff, OSD_WARNINGS_MAX_SIZE + 1, "BATT < FULL");
                 break;
             }
 
             // Visual beeper
             if (enabledWarnings & OSD_WARNING_VISUAL_BEEPER && showVisualBeeper) {
-                osdFormatMessage(buff, sizeof(buff), "  * * * *");
+                osdFormatMessage(buff, OSD_WARNINGS_MAX_SIZE + 1, "  * * * *");
                 break;
             }
 
-            return true;
```

Comment:
Was returning out of here previously unintended?

Reply:
Intended or not, it's inconsistent. It does not matter at the moment because we are blanking the screen before every redraw. But once we are moving to iterative updating, we _need_ to blank out the warnings space if no warning is active.